src/components/views/voip/CallView.tsx

```diff
@@ -522,41 +532,82 @@ export default class CallView extends React.Component<IProps, IState> {
+                // is it worth getting the size of the div to pass here?
                     this.props.call.getOpponentMember(), 1024, 1024, 'crop',
                 );
                 backgroundStyle.backgroundImage = 'url(' + backgroundAvatarUrl + ')';
                 onHoldBackground = <div className="mx_CallView_video_holdBackground" style={backgroundStyle} />;
-            }
-            if (!this.state.vidMuted) {
-                localVideoFeed = <VideoFeed type={VideoFeedType.Local} call={this.props.call} />;
-            }
 
-            contentView = <div className={containerClasses} ref={this.contentRef} onMouseMove={this.onMouseMove}>
-                {onHoldBackground}
-                <VideoFeed type={VideoFeedType.Remote} call={this.props.call} onResize={this.props.onResize} />
-                {localVideoFeed}
-                {holdTransferContent}
-                {callControls}
-            </div>;
-        } else {
-            const avatarSize = this.props.pipMode ? 76 : 160;
+                contentView = (
+                    <div className={containerClasses} ref={this.contentRef} onMouseMove={this.onMouseMove}>
+                        {onHoldBackground}
+                        {holdTransferContent}
+                        {callControls}
+                    </div>
+                );
+            } else {
+                const classes = classNames({
+                    mx_CallView_voice: true,
+                    mx_CallView_voice_hold: isOnHold,
+                });
+
+                contentView =(
+                    <div className={classes} onMouseMove={this.onMouseMove}>
+                        <div className="mx_CallView_voice_avatarsContainer">
+                            <div
+                                className="mx_CallView_voice_avatarContainer"
+                                style={{width: avatarSize, height: avatarSize}}
+                            >
+                                <RoomAvatar
+                                    room={callRoom}
+                                    height={avatarSize}
+                                    width={avatarSize}
+                                />
+                            </div>
+                        </div>
+                        {holdTransferContent}
+                        {callControls}
+                    </div>
+                );
+            }
+        } else if (this.props.call.noIncomingFeeds()) {
+            // Here we're reusing the css classes from voice on hold, because
+            // I am lazy. If this gets merged, the CallView might be subject
+            // to change anyway - I might take an axe to this file in order to
+            // try to get other things working
             const classes = classNames({
                 mx_CallView_voice: true,
-                mx_CallView_voice_hold: isOnHold,
             });
 
+            const feeds = this.state.feeds.map((feed, i) => {
```

Comment:
The local video feed - if there are no incoming feeds we, display connecting screen and look for local video feeds. Then we skip any that doesn't have video and we should be left with the local video feeds

Reply:
(we want to see our camera feed when we are connecting, I assume)

Reply:
but this returns nothing if `feed.isLocal()`?

Reply:
It will return nothing if the feed's video is muted and the feed is local - there is nothing to show. Does that make sense?

Reply:
Right, but surely this loop never returns anything, since it's inside the `this.props.call.noIncomingFeeds()` if block and skips local ones.

Reply:
We aren't skipping all local feeds, we are skipping the local feeds _only_ if they also have the video muted

(I am really sorry if I am missing something really obvious...)

Reply:
Ah! Sorry yes, that's what I was missing. I'm finding the API of listing all the feeds, local + remote, in the same list to be quite confusing and struggling to think when you'd ever really want to do that. Local & remote VideoFeed elements are instantiated the same way here but I think that's because of the antipattern of VideoFeed assuming 
that local/remote dictates the presentation of the feed (ie. that 'local' always means 'small').

Maybe we should be providing separate getters for the local vs remote feeds?

Reply:
> I'm finding the API of listing all the feeds, local + remote, in the same list to be quite confusing and struggling to think when you'd ever really want to do that.

This originates from [#16110](https://github.com/vector-im/element-web/issues/16110). Discord for example presents all streams in the same way no matter the source. I don't think having a separate `localFeeds` and `remoteFeeds` array would make things simpler but adding separate getters should :+1: